### PR TITLE
Rearrange folders according to Python packaging best practices

### LINKED ISSUE #249: Reorder directories to better conform to Python package standards
Right now we have the following directory structure:

```console
gcpy
    |
    gcpy
    |
    benchmark
    |       |
    |      modules
    |
    examples
            |
            several  other subdirectories not listed
```
But according to Python packaging best practice we should use a directory structure such as:
```console
gcpy
    |
    gcpy
       |
       benchmark
       |       |
       |       modules
       |
       examples
              |
              several other subdirectories not listed
```
Key features:
1. The root package folder `gcpy` now has a `src` folder, where all source code goes
2. `gcpy/gcpy` is now moved to `gcpy/src/gcpy`
3. All other folders (benchmark, examples) are now subfolders of `gcpy/src/gcpy`.

Background: comment by @lizziel in issue, see https://github.com/geoschem/gcpy/issues/245#issuecomment-1664514951

### PR DESCRIPTION
**Pull request by yantosca**
### Name and institution
Bob Yantosca
Harvard + GCST

### Overview

This PR (currently a draft) is the companion PR to #249.  We propose rearranging the GCPy folder structure from the current structure:
```console
gcpy
    |
    gcpy
    |
    benchmark
    |       |
    |      modules
    |
    examples
            |
            several  other subdirectories not listed
```
to:
```console
gcpy
    |
    gcpy
        |
        benchmark
        |       |
        |       modules
        |
        examples
               |
               several other subdirectories not listed
```

### Related issues
- Closes #249

### NOTES
- This is currently a draft PR and should not be merged.  Any feedback would be appreciated.  Tagging @lizziel @msulprizio @yidant @laestrada @SaptSinha

### CODE REVIEW

**Comment by yantosca** [comment]
I have made the following updates:
1. The `PYTHONPATH` variable can now be set in your .bashrc as follows:
    ```bash
    export PYTHONPATH=/path/to/gcpy
    ````

2. Moved `gcpy/benchmark` to gcpy/gcpy/benchmark`/

3. Moved `gcpy/examples` to `gcpy/gcpy/examples`.

4. Added `__init__.py` files in all subdirectories.

5. Make sure scripts are protected with an `if __name__ == "__main__":` block to prevent routines from executing upon import.

6. The `gcpy/benchmark/run_benchmark.py` script can now be invoked with:
    ```console
    from gcpy.benchmark import run_benchmark
    run_benchmark.main("1yr_fullchem_benchmark.yml")
    ```

7. The `gcpy/examples/diagnostics/compare_diags.py` script can now be invoked with:
    ```console
    from gcpy.examples import compare_diags
    compare_diags.main("input.yml")
    ```

**Comment by yantosca** [comment]
I have merged the updates from PR #251 (now in the `dev` branch) into the `feature/move-folders` branch.  This will allow us to develop on top of PR #251.

**Comment by yantosca** [comment]
Note. this should be merged after PR #246

**Comment by yantosca** [comment]
PR https://github.com/geoschem/gc-cloud-infrastructure/pull/51 should be merged concurrently with this PR, in order to avoid breaking the cloud benchmarking workflow.

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +41,12 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Routine `print_totals` now prints small and/or large numbers in scientific notation
 - Truncate names in benchmark & emissions tables to improve readability
 - Add TransportTracers species names to `gcpy/emissions_*.yml` files
+- Folder `gcpy/gcpy` is now  `gcpy/src/gcpy`, for adherence to Python packaging standards
```
Comment (msulprizio) [inline CHANGELOG.md:44]: Is this still the case or are we not using the `src` directory?

File: gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py [hunk 2]
```diff
@@ -388,7 +388,7 @@ def run_benchmark(config, bmk_year_ref, bmk_year_dev):
 
             # Create plots
             print("\nCreating plots for annual mean")
-            bmk.make_benchmark_conc_plots(
+            bmkf.make_benchmark_conc_plots(
```
Comment (msulprizio) [inline gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py:391]: I think this is a typo - `bmk` is still used above

**Review by msulprizio — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (yantosca) [reply CHANGELOG.md:44]: Nice catch @msulprizio.  I'll edit the changelog.  If we use `gcpy/src` then we'd have to add `path/to/gcpy/src` to `PYTHONPATH`, which is a little ugly.

File: gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py (hunk 2 above)
Comment (yantosca) [reply gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py:391]: Thanks @msulprizio.  I'll fix this.

File: CHANGELOG.md (hunk 1 above)
Comment (yantosca) [reply CHANGELOG.md]: Now removed in commit cf6fc12

File: gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py (hunk 2 above)
Comment (yantosca) [reply gcpy/benchmark/modules/run_1yr_fullchem_benchmark.py]: Now fixed in commit cf6fc12

**Comment by yantosca** [comment]
@msulprizio, I will merge this locally and make sure I can plot benchmarks before pushing.

**Review by msulprizio — APPROVED** [review]

**Pull request merged**